torch/csrc/jit/codegen/cuda/ir_internal_nodes.h

```diff
@@ -544,6 +544,15 @@ class TORCH_CUDA_CU_API IterDomain : public Val {
     }
   }
 
+  // Signature that's easier to use for copying from one ID to another.
+  void padToMultipleOfWarp(c10::optional<int64_t> maybe_to_size) {
+    if (maybe_to_size.has_value()) {
+      padToMultipleOfWarp(maybe_to_size.value());
+    } else {
+      padToMultipleOfWarp();
+    }
+  }
+
```

Comment:
This parameter is used to specify if we want to pad to a statically available size or we want the kernel_executor to figure out what size to pad it to. 

When the size is statically constant there's opportunities for some optimization. Currently there's only one, which is fusing warp broadcast with warp reduce for single warp. We might have more.

Reply:
Yeah, the behavior sounds reasonable. I'm just curious if we need to have the version with `int` parameter and another with `c10::optional<int64_t>`, which is added in this PR. It's just about a small design difference.

Reply:
Ah I see. Sorry I read the question incorrectly.

I think the original interface (the int one) could be deprecated since optional is cleaner and int64_t would be more compatible with how we handle integers in other parts.

I could do a follow up PR if needed :).

Reply:
Cleaned this up